File: examples/python/parsed_messages.py
import mcapable

from common import write_sample_file


def main() -> None:
    path = write_sample_file()
    reader = mcapable.Reader.from_path(str(path))

    print("Parsed Message Stream Example")
    stream = reader.messages()
    parsed = stream.parsed()
    for i, item in enumerate(parsed):
        if isinstance(item, (dict, list)):
            print(f"Message {i}: parsed JSON {item}")
        else:
            print(f"Message {i}: raw bytes size={len(item)}")
        if i >= 4:
            break

    reader = parsed.into_reader()
    print(f"Total messages: {sum(1 for _ in reader.messages())}")


if __name__ == "__main__":
    main()

File: examples/typescript/parsed_messages.ts
import { Reader } from "@mcapable/wasm";

import { sampleBytes } from "./common";

function main(): void {
  const reader = Reader.fromBytes(sampleBytes());
  const stream = reader.messages();
  const parsed = stream.parsed();
  let count = 0;
  for (;;) {
    const item = parsed.next();
    if (!item) break;
    if (item instanceof Uint8Array) {
      console.log(`Message ${count}: raw bytes size=${item.length}`);
    } else {
      console.log(`Message ${count}: parsed JSON`, item);
    }
    if (++count >= 5) {
      break;
    }
  }
}

main();
